src/main/java/focusedCrawler/link/classifier/LinkClassifierAuthority.java

```diff
@@ -53,9 +52,11 @@ public LinkClassifierAuthority(LinkNeighborhoodWrapper wrapper,String[] attribut
 		        	  if(!page.getURL().getHost().equals(url.getHost())){
 		        		  Instance instance = entry.getValue();
 		        		  double[] values = instance.getValues();
-		        		  weka.core.Instance instanceWeka = new weka.core.Instance(1, values);
-		        		  instanceWeka.setDataset(instances);
-		        		  double[] prob = classifier.distributionForInstance(instanceWeka);
+//		        		  weka.core.Instance instanceWeka = new weka.core.Instance(1, values);
+//		        		  instanceWeka.setDataset(instances);
+//		        		  double[] prob = classifier.distributionForInstance(instanceWeka);
+		        		  double[] prob = new double[2];
+		        		  int predictedValue = ((SVM) classifier).predict(values, prob);

```

Comment:
You should be using SoftClassifier interface instead of Classifier, then you wouldn't need to do a cast, and the code would work with any classifier algorithm implementation. Down casting is usually a code smell.

Reply:
I will change that and test again.